core/rpc/json/openrpc/reflect.go

```diff
@@ -0,0 +1,238 @@
+package openrpc
+
+import (
+	"cmp"
+	"math/big"
+	"reflect"
+	"slices"
+	"strings"
+	"unicode/utf8"
+
+	"github.com/kwilteam/kwil-db/core/types"
+)
+
+type MethodDefinition struct {
+	Description  string
+	Summary      string
+	RequestType  reflect.Type
+	ResponseType reflect.Type
+	RespTypeDesc string
+	// Errors ...
+}
+
+// InventoryAPI iterates through all handler types, making a map of all unique
+// OpenRPC schemas and building a []Method with every Param populated and
+// referencing a Schema if an "object" instead of a plain type.
+func InventoryAPI(handlerDefs map[string]*MethodDefinition, knownSchemas map[reflect.Type]Schema) []Method {
```

Comment:
mmm, now I'm thinking to move this `openrpc` package to internal, at least for now.  Initially I had the service specs generated from types and such in `core`, but now I have the RPC server's generating them based on the services that are registered with them... and that's the only things using the types and functions defined here.